src/control/server/mgmt_storage.go

```diff
@@ -0,0 +1,246 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package main
+
+import (
+	"github.com/daos-stack/daos/src/control/common"
+	pb "github.com/daos-stack/daos/src/control/common/proto/mgmt"
+	"github.com/daos-stack/daos/src/control/log"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// addState creates, populates and returns ResponseState in addition
+// to logging any err.
+func addState(
+	status pb.ResponseStatus, errMsg string, infoMsg string, logDepth int,
+	contextMsg string) *pb.ResponseState {
+
+	state := new(pb.ResponseState)
+
+	state.Status = status
+	state.Error = errMsg
+	state.Info = infoMsg
+
+	if errMsg != "" {
+		log.Errordf(logDepth, contextMsg+": "+errMsg)
+	}
+
+	return state
+}
+
+// ScanStorage discovers non-volatile storage hardware on node.
+func (c *controlService) ScanStorage(
+	ctx context.Context, params *pb.ScanStorageParams) (
```

Comment:
Are ctx and params going to be used?

Reply:
ctx and params have to be provided to match the proto service definition, params might be used in future to filter devices

Reply:
Ah, okay. Got it.